### DISCUSSION TITLE
No module named 'miplearn.instance.base'; 'miplearn.instance' is not a package

### DISCUSSION THREAD

**Discussion by HuigenYe** [ANL-CEEESA/MIPLearn#8] [Help & FAQ] [answered]
Hi iSoron, when I use MIPlearn to run the code "tsp.py", it tells me that "No module named 'miplearn.instance.base'; 'miplearn.instance' is not a package". Is this because I made an error installing miplearn? Thank you for your reply.

**Comment by iSoron** [comment, marked as answer]
Hi @qwqjvruo, this is probably a package installation issue. Could you try the following:
```
# Create a new conda environment
$ conda create -n miplearn-test python=3.7

# Activate environment
$ conda activate miplearn-test

# Install miplearn
$ pip install miplearn

# Launch python and access TSP problem generator
$ python
>>> from miplearn.problems.tsp import TravelingSalesmanGenerator
```

**Reply by HuigenYe** [reply]
Hi iSoron,  thank you for your reply. After using Python version 3.7 instead of 3.10, the problem was indeed solved. Thank you very much for your answer!
